lib/libzrepl/mgmt_conn.c

```diff
@@ -955,6 +955,11 @@ uzfs_zvol_execute_async_command(void *arg)
 		rc = uzfs_zvol_create_snapshot_update_zap(zinfo, snap,
 		    async_task->hdr.io_seq);
 		if (rc != 0) {
+			/*
+			 * snap create command failed, close the io
+			 * connection so that it can start the rebuilding
+			 */
+			VERIFY0(shutdown(zinfo->io_fd, SHUT_RDWR));
```

Comment:
this doesn't seem to be safe yet. We can't be sure what the other threads are doing here, and what the status of io_fd

Reply:
for SNAP_CREATE, it is safe as no IOs will be happening.